### PR TITLE
HTTP Logging

### PR DESCRIPTION
**Pull request by lnkuiper**
This PR implements a basic version of HTTP logging.

It can be toggled like so:
```sql
SET enable_http_logging=true;
SET enable_http_logging=false;
```

By default, the headers of HTTP requests will be logged to stdout, but can also be appended to a file:
```sql
SET http_logging_output='httplog.txt';
```

The output looks like this:
```
HTTP Request:
    <header key>: <header value>

HTTP Response:
    <header key>: <header value>
```
This can be useful when tracking down connection errors.

### CODE REVIEW

File: src/include/duckdb/main/client_config.hpp [hunk 1]
```diff
@@ -115,6 +117,12 @@ struct ClientConfig {
 	//! Defaults to PhysicalMaterializedCollector
 	get_result_collector_t result_collector = nullptr;
 
+	//! If HTTP logging is enabled or not.
+	bool enable_http_logging = false;
```
Comment (Tishj) [inline src/include/duckdb/main/client_config.hpp:121]: Shouldnt these be extension settings added by httpfs?

Comment (lnkuiper) [reply src/include/duckdb/main/client_config.hpp:121]: Extension installs go through httplib before httpfs is even installed, so if we want to be able to log those, we can't add this setting in httpfs, that's too late

**Review by samansmink — APPROVED** [review]
LGTM!

**Comment by samansmink** [comment]
CI should probably rerun for `Python / Linux Extensions` and the Linux HTTPFS one looks like it might be related to this PR?

**Comment by lnkuiper** [comment]
Thanks for the review! Re-running CI, will look into any issues that pop up

**Comment by samansmink** [comment]
Just popped into my head: I'm working on bearer token support and realized that this feature would print out its secrets very easily and in clear text. We could consider only allowing enabling this option when `allow_unredacted_secrets` is set?

Of course on the other hand it would not be very hard to retrieve the bearer token secret through some other path such as querying some url you control.

**Comment by lnkuiper** [comment]
Sure, that sounds very reasonable. Let's discuss tomorrow!

**Comment by lnkuiper** [comment]
Sam and I discussed and concluded that the current behavior regarding `allow_unredacted_secrets` is fine, so this is ready to merge once CI passes.

**Comment by lnkuiper** [comment]
@Mytherin I think this is ready to go!

**Pull request merged**